src/archival/listener.rs

```diff
@@ -12,23 +16,36 @@ pub async fn listen(pool: PgPool) -> Result<(), Error> {
             println!("Notification Payload: {}", notification.payload());
             let payload: InternetArchiveUrls =
                 serde_json::from_str(notification.payload()).unwrap();
-            archive(
+            if let Err(e) = archive(
                 payload.id,
                 payload.url.unwrap(),
                 payload.retry_count.unwrap(),
                 &pool,
             )
-            .await;
+            .await
+            {
+                eprintln!("Archival Error: {}", e)
+            }
```

Comment:
Those patterns are bad for readability
```
if f(
...
)
.await
{ 
    ...
}
```

it's easy to forget that we are doing an `if` or `match` after the `.await`; It's better to separate the future's declaration and the check:
```rust
let archive_future =  archive(
                payload.id,
                payload.url.unwrap(),
                payload.retry_count.unwrap(),
                &pool,
            );

if let Err(e) = archive_future.await {
                eprintln!("Archival Error: {}", e)
            }
```

Reply:
For cases like these, I am thinking to let the error propagate upwards, and instead of just printing, it would be better to use `?` and handle all the errors at a central place. This is what I am thinking after I discovered sentry.
For the cases when I know what should be done in case of Error, I can handle them with match statements, otherwise, I think propagating above and let sentry send the appropriate error seems right to me. 
@RustyNova016 does it seem better?

Reply:
I would definitly propagate them up to the point of the task being spawned. Like for the Poller, I would check them after calling the loop function.

But remember that you don't have to check everything. You just need to check what you can control. Cannot connect to the database? Can't do anything to help. IA return a 400? Maybe you can try something else.

Reply:
Yupp. When I know the error, in those cases I am thinking to handle the error. Otherwise I am propagating above